src/VitessceGrid.js

```diff
@@ -36,106 +35,98 @@ function shallowEqual(objA, objB) {
+    const Component = getComponent(v.component);
+    const onReady = () => {
+      readyComponentKeys.add(k);
+      if (readyComponentKeys === Object.keys(gridComponents).length) {
+        // The sets are now equal
+        onAllReady();
+      }
+    };
+
+    const removeGridComponent = () => {
+      const newGridComponents = { ...gridComponents };
+      delete newGridComponents[k];
+      setGridComponents(newGridComponents);
+    };
 
-    const layoutChildren = Object.entries(components).map(([k, v]) => {
-      const Component = getComponent(v.component);
-      const onReady = () => {
-        this.setState((state) => {
-          const { readyComponentKeys } = state;
-          const newReadyComponentKeys = new Set(readyComponentKeys);
-          newReadyComponentKeys.add(k);
-          if (newReadyComponentKeys.size === Object.keys(components).length) {
-            // The sets are now equal.
-            onAllReady();
-          }
-          return { readyComponentKeys: newReadyComponentKeys };
-        });
-      };
-      return (
-        <div key={k}>
-          <Component {... v.props} onReady={onReady} />
-        </div>
-      );
-    });
-
-    const maxRows = getMaxRows(layouts);
     return (
-      <React.Fragment>
-        {style}
-        <ResponsiveGridLayout
-          className="layout"
-          cols={cols}
-          layouts={layouts}
-          breakpoints={breakpoints}
-          rowHeight={
-            rowHeight
-            || (
-              (window.innerHeight - 2 * padding - (maxRows - 1) * margin)
-              / maxRows
-            )}
-          containerPadding={[padding, padding]}
-          margin={[margin, margin]}
-          draggableHandle={draggableHandle}
-          {... reactGridLayoutProps}
-        >
-          {layoutChildren}
-        </ResponsiveGridLayout>
-      </React.Fragment>
+      <div key={k}>
+        <Component
+          {... v.props}
+          removeGridComponent={removeGridComponent}
+          onReady={onReady}
+        />
+      </div>
     );
-  }
-}
+  });
+  const maxRows = getMaxRows(layouts);
+  return (
+    <React.Fragment>
+      {style}
+      <ResponsiveGridLayout
+        className="layout"
+        cols={cols}
+        layouts={layouts}
+        breakpoints={breakpoints}
+        rowHeight={
+          rowHeight
+          || (
+            (window.innerHeight - 2 * padding - (maxRows - 1) * margin)
+            / maxRows
+          )}
+        containerPadding={[padding, padding]}
+        margin={[margin, margin]}
+        draggableHandle={draggableHandle}
+        {... reactGridLayoutProps}
+      >
+        {layoutChildren}
+      </ResponsiveGridLayout>
+    </React.Fragment>
+  );
+};
 
 VitessceGrid.defaultProps = {
   padding: 10,
   margin: 10,
   onAllReady: () => {},
 };
+
+export default React.memo(
+  VitessceGrid,
+  (prevProps, nextProps) => !shallowEqual(prevProps, nextProps),
```

Comment:
@manzt : Not related to the main problem, but is the negation correct? https://reactjs.org/docs/react-api.html#reactmemo :
> Unlike the shouldComponentUpdate() method on class components, the areEqual function returns true if the props are equal and false if the props are not equal. This is the inverse from shouldComponentUpdate.

Reply:
I think you're right, @mccalluc. good catch